common/v2/features/ProtectTransaction/components/WithProtectTransaction.tsx

```diff
@@ -0,0 +1,234 @@
+  position: relative;
+  flex: 0 0 100%;
+  width: 100%;
+  max-width: 100%;
+
+  ${({ protectTxShown }) =>
+    protectTxShown &&
+    `
+    @media (min-width: ${BREAK_POINTS.SCREEN_SM}) {
+      flex: 0 0 calc(100vw - 375px - 4.5rem);
+      width: calc(100vw - 375px - 4.5rem);
+      max-width: calc(100vw - 375px - 4.5rem);
+    }
+  `};
+
+  @media (min-width: ${BREAK_POINTS.SCREEN_MD}) {
+    flex: 0 0 calc(650px - 4.5rem);
+    width: calc(650px - 4.5rem);
+    max-width: calc(650px - 4.5rem);
+  }
+`;
+
+const WithProtectTransactionSide = styled.div`
+  position: absolute;
+  left: 50%;
+  transform: translateX(-50%);
+  width: 375px;
+  min-width: 375px;
+  max-width: 100vw;
+  margin-top: calc(-1.5rem - 44px - 15px);
+
+  @media (min-width: ${BREAK_POINTS.SCREEN_SM}) {
+    position: initial;
+    width: 375px;
+    margin-top: -1.5rem;
+    margin-left: 2.25rem;
+    border-left: 15px solid ${COLORS.BG_GRAY};
+    min-height: calc(100% + 115px);
+    transform: unset;
+
+    section {
+      position: relative;
+      height: 100%;
+      padding: 30px 15px 15px;
+      box-shadow: none;
+    }
+  }
+`;
+
+interface WithProtectTransactionProp extends IStepComponentProps {
+  withProtectApi?: WithProtectApiFactory;
+  customDetails?: JSX.Element;
+  protectTxButton?(): JSX.Element;
+}
+
+export function withProtectTransaction(
+  WrappedComponent: React.ComponentType<WithProtectTransactionProp>,
+  SignComponent: React.ComponentType<IStepComponentProps>
+) {
+  return function WithProtectTransaction({
+    txConfig: txConfigMain,
+    signedTx: signedTxMain,
+    txReceipt: txReceiptMain,
+    onComplete: onCompleteMain,
+    withProtectApi,
+    customDetails,
+    resetFlow
+  }: WithProtectTransactionProp) {
+    const {
+      handleProtectedTransactionSubmit,
+      handleProtectedTransactionConfirmAndSend,
+      protectedTransactionTxFactoryState
+    } = useStateReducer(ProtectedTxConfigFactory, {
+      txConfig: protectedTxConfigInitialState,
+      txReceipt: null
+    });
+
+    const {
+      withProtectState: { protectTxShown, stepIndex, protectTxEnabled, isWeb3Wallet },
+      handleTransactionReport,
+      goOnNextStep,
+      goOnInitialStepOrFetchReport,
+      formCallback,
+      showHideTransactionProtection
+    } = withProtectApi!;
+
+    const isMdScreen = useMediaQuery(`(min-width: ${BREAK_POINTS.SCREEN_MD})`);
+
+    const onModalBackdropClick = useCallback(
+      e => {
+        e.preventDefault();
+
+        if (showHideTransactionProtection) {
+          showHideTransactionProtection(false);
+        }
+      },
+      [showHideTransactionProtection]
+    );
+
+    const onMobileShowReportClick = useCallback(
+      e => {
+        e.preventDefault();
+
+        if (showHideTransactionProtection) {
+          showHideTransactionProtection(true);
+        }
+      },
+      [showHideTransactionProtection]
+    );
+
+    return useMemo(
```

Comment:
I want to limit the rerenders to only used props. A lot unnecessary rerenders can happen, and can be costly because main and side component will rerender.